src/data/reducers/shopify.js

```diff
@@ -91,24 +106,42 @@ const getProduct = id => {
 // Gets all the collections from Shopify
 const getCollections = () => {
   return dispatch => {
+    dispatch({ type: COLLECTIONS_FETCHING });
     client.collection.fetchAllWithProducts().then(res => {
       dispatch({
-        type: COLLECTION_FOUND,
+        type: COLLECTIONS_FOUND,
         payload: res,
       });
     });
   };
 };
 
+const createCheckout = dispatch => {
+  client.checkout.create().then(res => {
+    dispatch({
+      type: CHECKOUT_FOUND,
+      payload: res,
+    });
+  });
+};
+
 // Creates initial checkout state from Shopify
-const checkout = () => {
+const initCheckout = () => {
   return dispatch => {
-    client.checkout.create().then(res => {
-      dispatch({
-        type: CHECKOUT_FOUND,
-        payload: res,
-      });
-    });
+    dispatch({ type: CHECKOUT_FETCHING });
+    if (CHECKOUT_ID_FROM_LOCAL_STORAGE) {
+      client.checkout
+        .fetch(CHECKOUT_ID_FROM_LOCAL_STORAGE)
+        .then(res =>
+          dispatch({
+            type: CHECKOUT_FOUND,
+            payload: res,
+          }),
+        )
+        .catch(err => createCheckout(dispatch));
```

Comment:
fetch checkout using ID from local storage if it exists, instead of creating a new one.